File: iOSTest/Models/Message.m
//
//  Message.m
//  iOSTest
//
//  Created by HannDev LLC on 9/9/19.
//  Copyright © 2019 HannDev LLC. All rights reserved.
//

#import "Message.h"

@implementation Message 

- (instancetype)initWithDictionary:(NSDictionary *)dictionary
{
    self = [super init];
    if (self)
    {
        self.userID = dictionary[@"user_id"];
        self.username = dictionary[@"username"];
        self.avatarURL = [NSURL URLWithString:dictionary[@"avatar_url"]];
        self.text = dictionary[@"message"];
    }
    
    return self;
}

- (instancetype)initWithTestName:(NSString *)name withTestMessage:(NSString *)message
{
    self = [super init];
    if (self)
    {
        self.userID = 0;
        self.username = name;
        self.avatarURL = [NSURL URLWithString:@"https://upload.wikimedia.org/wikipedia/commons/thumb/8/85/Smiley.svg/220px-Smiley.svg.png"];
        self.text = message;
    }
    
    return self;
}

@end

File: iOSTest/Message.swift
//
//  Message.swift
//  iOSTest
//
//  Created by Greg Hughes on 2/6/20.
//  Copyright © 2020 D&ATechnologies. All rights reserved.
//

import Foundation


struct TopLevelDictionary : Codable {
    
    var data : [Message]
    
}

struct Message : Codable {
    var userID : String
    var username : String
    var avatarURL : URL
    var text : String
    
    init(testName : String, withTestMessage: String) {
        self.username = testName
        self.text = withTestMessage
        self.avatarURL = URL(string: "https://upload.wikimedia.org/wikipedia/commons/thumb/8/85/Smiley.svg/220px-Smiley.svg.png")!
        self.userID = "0"
    }
    
    enum CodingKeys: String,CodingKey {
        case userID = "user_id"
        case username = "name"
        case avatarURL = "avatar_url"
        case text = "message"
    }
}
